### DISCUSSION TITLE
Get user input inside of a shell script

### DISCUSSION THREAD

**Discussion by tdaron** [helix-editor/helix#9174] [Q&A] [answered]
Hey ! 

I'm currently having a **very simple** script `man.sh` containing:
```sh
read manpage
man $manpage | cat
```

And a helix shortcut:

`"backspace" = {l=[":hnew", ":insert-output sh ./man.sh","collapse_selection"]}`

but i can't figure how to get the `read` working (to get which manpage user wanna see)

Any ideas ? Is it even possible ?

(and how do i tell helix the new buffer is readonly and does not need to be saved ?)

**Comment by tdaron** [comment]
The only program i could find that do that is [lf](https://github.com/gokcehan/lf)

when i do :insert-output lf , it show me the filepicker and i can choose a file, but i couldn't manage to make a custom bash script retrieving user input works inside of helix

:insert-output read -r something

is not working, and read exits with status 1 which means it instantly get EOF, why is helix sending EOF to read ?

**Comment by kirawi** [comment]
https://github.com/helix-editor/helix/issues/1976

**Reply by tdaron** [reply]
Thanks ! I will try something using wezterm then !

But do you know why especially `lf` works the way i want without any trick ? i just ran insert-output lf and it works the way i want

**Reply by gabydd** [reply, marked as answer]
Was looking into this and lf uses /dev/tty

**Reply by tdaron** [reply]
thanks
